### ISSUE TITLE
Add assertions for promise values

### ISSUE THREAD

**Issue by arikon** [moll/js-must#22]
As in should.js

```js
return expect(promise).to.eventually.be.true();
return expect(promise).to.eventually.be.rejectedWith(Error);
// etc
```

Or something like in Chai.js as Promised: http://chaijs.com/plugins/chai-as-promised

**Comment by madarche** [comment]
+1 !

**Comment by JohnnyEstilles** [comment]
+1

**Comment by mikeerickson** [comment]
+1

**Comment by moll** [comment]
I had an idea on how to add promise support wholesale to every current and future matcher, and while testing it out, I accidentally ended up doing it. :innocent: 

I'm open for naming suggestions, but with the latest master (use `npm i git+https://github.com/moll/js-must.git` to try) you should (or need I say _must_ ^_^) be able to do:

```javascript
it("must pass", function() {
  return Promise.resolve(42).must.then.equal(42)
})
```

So, rather than separate matchers for asserting on resolves values, [`Must.prototype.then`](https://github.com/moll/js-must/blob/master/doc/API.md#Must.prototype.then) adds promise support to every matcher. 

I've added `eventually` as an alias to `then`. Also thinking of adding `soon` as an alias. Any proposals? :-)

Please do give it a go and share your comments. Thank you!

**Comment by arikon** [comment]
@moll It's a good solution! :+1:

**Comment by moll** [comment]
FYI to all, I released v0.13.0-beta1 as a stop-gap before the next stable. If any of you are in a hurry with the promise stuff, go ahead. ;-)

**Comment by moll** [comment]
I'll close this issue as v0.13 has been released since. ;-)
